networkx/algorithms/centrality/trophic.py

```diff
@@ -76,7 +76,7 @@ def trophic_levels(G, weight="weight"):
     # all other nodes have levels as calculated
     nonzero_node_ids = (node_id for node_id, degree in G.in_degree if degree != 0)
     for i, node_id in enumerate(nonzero_node_ids):
-        levels[node_id] = y[i]
+        levels[node_id] = float(y[i])
```

Comment:
`array.item` is a good find @dschult! For cases like this, would you rather do `float(y[i])` or `y.item(i)`?

Reply:
Because `y` is a numpy array, I think it should be `y.item(i)`. And its faster. It might not be as clear unless people know what the `item` method is. They have to look it up. But the same is true for `tolist`. We'll just make them look it up. :\

For speeds:  setup `b=np.arange(1000);labels={}` then inside loop: `for i, v in enumerate(b):`
`labels[v] = float(v)`        80.3 µs ± 137 ns
`labels[v] = b.item(i)`      98.8 µs ± 103 ns
`labels[v] = float(b[i])`    126 µs ± 28 ns
`labels[v] = v.item()`       252 µs ± 1.68 µs
 
Anyway, I also like `y.item(i)` better because it chooses to make python `int`s when it can.